### DISCUSSION TITLE
Accessing yourls_link() from another php page.

### DISCUSSION THREAD

**Discussion by hongsolo9** [YOURLS/YOURLS#3466] [General]
Hi all, first of all many thanks to @ozh, @dgw and @LeoColomb for YOURLS. My brain is about to be fried, therefore seeking help here.

I have a a YOURLS installation running fine. I have another (non-yourls) php page which i want to access the yourls_link() function from. But I'm getting errors in Apache error_log saying yours_link() is an undefined function from this (non-yourls) php page.

I'm sure this is real simple but my brain has stopped working...

**Comment by dgw** [comment]
It is probably easier to have your other page call the `expand` [action](https://yourls.org/#API) of the [YOURLS API](https://docs.yourls.org/guide/advanced/passwordless-api.html) than to `require()` all the things that would be necessary to make a function work directly in PHP, if I'm honest.

@YOURLS/core The API endpoints and stuff documented at https://yourls.org/#API probably need to be included in the docs page/section about the Passwordless API. Agree?

**Reply by LeoColomb** [reply]
> probably need to be included in the docs page/section

I agree 😊

**Reply by hongsolo9** [reply]
Thank you @dgw and @LeoColomb ! This is interesting - I got to read more into the docs. Thanks for this!

**Comment by ozh** [comment]
Short way : `<?php require_once '/path/to/yourls_root/includes/load-yourls.php'; ?>`
That's how the YOURLS engine starts, see for instance https://github.com/YOURLS/YOURLS/blob/master/sample-public-front-page.txt

More satisfying way, depending on what you need to achieve : using the YOURLS API as @dgw said

**Reply by hongsolo9** [reply]
Got it, @ozh ! Worked beautifully!
